compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -465,6 +482,55 @@ object desugar {
       case _ =>
         (Nil, tree)
 
+  private def referencesName(vdef: ValDef, names: Set[TermName])(using Context): Boolean =
+    vdef.tpt.existsSubTree:
+      case Ident(name: TermName) => names.contains(name)
+      case _ => false
+
+  /** Fit evidence `params` into the `mparamss` parameter lists */
```

Comment:
It would be good to describe this method a bit more. I'd add a description with `@param` and `@return` what the parameters are and what gets returned.